docs/rfcs/autoops-flag-trigger.md

```diff
@@ -0,0 +1,228 @@
+    `deleted`               tinyint(1) NOT NULL DEFAULT '0',
+    `created_at`            bigint       NOT NULL,
+    `updated_at`            bigint       NOT NULL,
+    PRIMARY KEY (`id`)
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
+```
+
+Table explanation:\
+`id`: The unique ID of the flag trigger.\
+`feature_id`: The ID of the feature. \
+`environment_namespace`: The namespace of the environment.\
+`name`: The name of the flag trigger.\
+`type`: The type of the flag trigger, it can be **webhook**, etc.\
+`action`: The action of the flag trigger, it can be turned on(1)/off(0) the feature flag.\
+`description`: The description of the flag trigger.\
+`trigger_times`: The times of the flag trigger has been triggered. \
+`last_triggered_at`: The last time of the flag trigger has been triggered.\
+`uuid`: The uuid of the flag trigger. It will be used to create and reset the trigger url.\
+`disabled`: The flag trigger is disabled or not.\
+`deleted`: The flag trigger is deleted or not.
+
+* Reuse the current implementation to encrypt/decrypt the URL.
+
+### AutoOps Server
+
+* Add new API to manage the flag triggers.
+* As we are supporting webhook trigger first, but in the feature we will support more trigger types, so we need to
+  design the trigger architecture to support more trigger types. But one thing is that we should make the trigger
+  architecture simple and easy to use.
+
+#### AutoOps Flag Trigger API
+
+- Create a new flag trigger
+
+```Grpc
+rpc CreateFlagTrigger(CreateFlagTriggerRequest) returns (CreateFlagTriggerResponse) {}
+```
+
+```proto
+message CreateFlagTriggerRequest {
+    string feature_id = 1;
+    string environment_namespace = 2;
+    string name = 3;
+    int32 type = 4;
+    int32 action = 5;
+    string description = 6;
+}
+```
+
+- Update a flag trigger
+
+This update API may not be provided to the user. Because the real use case is create a new flag trigger and then reset
+the trigger url or just delete the flag trigger.
+
+```Grpc
+rpc UpdateFlagTrigger(UpdateFlagTriggerRequest) returns (UpdateFlagTriggerResponse) {}
+```
+
+```proto
+message UpdateFlagTriggerRequest {
+    string id = 1;
+    string feature_id = 2;
+    string environment_namespace = 3;
+    string name = 4;
+    int32 type = 5;
+    int32 action = 6;
+    string description = 7;
+}
+```
+
+- Reset a flag trigger
+
+```Grpc
+rpc ResetFlagTrigger(ResetFlagTriggerRequest) returns (ResetFlagTriggerResponse) {}
+```
+
+```proto
+message ResetFlagTriggerRequest {
+    string id = 1;
+}
+```
+
+- Disable a flag trigger
+
+```Grpc
+rpc DisableFlagTrigger(DisableFlagTriggerRequest) returns (DisableFlagTriggerResponse) {}
+```
+
+```proto
+message DisableFlagTriggerRequest {
+    string id = 1;
+}
+```
+
+- Delete a flag trigger
+
+```Grpc
+rpc DeleteFlagTrigger(DeleteFlagTriggerRequest) returns (DeleteFlagTriggerResponse) {}
+```
+
+```proto
+message DeleteFlagTriggerRequest {
+    string id = 1;
```

Comment:
Just to let you know, all requests contain a command.
But this minor detail doesn't need to be included in the RFC.
We want to ensure that the whole flow is correct. Which it is.

Reference.
https://github.com/bucketeer-io/bucketeer/blob/main/proto/experiment/service.proto#L191-L195

Reply:
Yeah, I have noticed that the command patten in the project.